standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/metasummary/MetadataTableSummary.java

```diff
@@ -0,0 +1,249 @@
+  }
+
+  public int getColCount() {
+    return colCount;
+  }
+
+  public void setColCount(int colCount) {
+    this.colCount = colCount;
+  }
+
+  public int getPartitionColumnCount() {
+    return partitionColumnCount;
+  }
+
+  public void setPartitionColumnCount(int partitionColumnCount) {
+    this.partitionColumnCount = partitionColumnCount;
+  }
+
+  public int getPartitionCount() {
+    return partitionCount;
+  }
+
+  public void setPartitionCount(int partitionCount) {
+    this.partitionCount = partitionCount;
+  }
+
+  public long getTotalSize() {
+    return totalSize;
+  }
+
+  public void setTotalSize(long totalSize) {
+    this.totalSize = totalSize;
+  }
+
+  public long getNumRows() {
+    return numRows;
+  }
+
+  public void setNumRows(long numRows) {
+    this.numRows = numRows;
+  }
+
+  public long getNumFiles() {
+    return numFiles;
+  }
+
+  public void setNumFiles(long numFiles) {
+    this.numFiles = numFiles;
+  }
+
+  public String getTableType() {
+    return tableType;
+  }
+
+  public void setTableType(String tableType) {
+    this.tableType = tableType;
+  }
+
+  public String getFileFormat() {
+    return fileFormat;
+  }
+
+  public void setFileFormat(String fileFormat) {
+    this.fileFormat = fileFormat;
+  }
+
+  public String getCompressionType() {
+    return compressionType;
+  }
+
+  public void setCompressionType(String compressionType) {
+    this.compressionType = compressionType;
+  }
+
+  public void setArrayColumnCount(int arrayColumnCount) {
+    this.arrayColumnCount = arrayColumnCount;
+  }
+
+  public int getArrayColumnCount() {
+    return arrayColumnCount;
+  }
+
+  public void setStructColumnCount(int structColumnCount) {
+    this.structColumnCount = structColumnCount;
+  }
+
+  public int getStructColumnCount() {
+    return structColumnCount;
+  }
+
+  public void setMapColumnCount(int mapColumnCount) {
+    this.mapColumnCount = mapColumnCount;
+  }
+
+  public int getMapColumnCount() {
+    return mapColumnCount;
+  }
+
+  public Map<String, Object> getExtraSummary() {
+    return summary;
+  }
+
+  public MetadataTableSummary addExtra(String key, SummaryMapBuilder builder) {
+    if (summary == null) {
+      summary = new HashMap<>();
+    }
+    summary.put(key, builder.build());
+    return this;
+  }
+
+  public void addExtra(SummaryMapBuilder builder) {
+    if (summary == null) {
+      summary = new HashMap<>();
+    }
+    summary.putAll(builder.build());
+  }
+
+  @Override
+  public String toString() {
+    return "TableSummary {" + "owner='" + owner + '\'' + ", db_name='" + dbName + '\'' + ", table_name='" + tblName +
+        '\'' + ", column_count=" + colCount + ", partition_count=" + partitionCount + ", table_type='" + tableType + '\'' +
+        ", file_format='" + fileFormat + '\'' + ", compression_type='" + compressionType + ", numRows=" + numRows +
+        ", numFiles=" + numFiles + ", size_bytes=" + totalSize + ", partition_column_count=" + partitionColumnCount +
```

Comment:
Should we add a boolean field `hasPartitions` and then add partitionCount and partitionColumnCount depending on this boolean count.

Reply:
IMO it is ok for `toString` method, it's here just for debugging.